tap_klaviyo/streams.py

```diff
@@ -75,12 +82,94 @@ def post_process(
+        """Return a context dictionary for child streams."""
+        return {
+            "campaign_id": record["id"],
+            "campaign_status": record["attributes"]["status"],
+        }
+
+    def parse_response(self, response: requests.Response) -> t.Iterable[dict]:
+        self.process_included(response)
+        yield from super().parse_response(response)
+
+    def process_included(self, response: requests.Response):
+        self.included_map = {included['id']: included for included in
+                             extract_jsonpath(self.included_jsonpath, input=response.json())}
+
+
+class CampaignValuesReportsStream(KlaviyoStream):
+    name = "campaign_values_reports"
+    path = "/campaign-values-reports"
+    rest_method = "POST"
+    primary_keys = ["id"]
+    replication_key = None
+    parent_stream_type = CampaignsStream
+    schema_filepath = SCHEMAS_DIR / "campaign_values_reports.json"
+    backoff_max_tries = 10
+
+    def post_process(self, row: dict, context: dict | None = None) -> dict | None:
+        row["campaign_id"] = context["campaign_id"]
+        row["campaign_status"] = context["campaign_status"]
+        row["generated_at"] = datetime.now().isoformat()
+        return row
+
+    def prepare_request_payload(
+            self,
+            context: dict | None,
+            next_page_token: t.Optional[t.Any],
+    ) -> dict | None:
+        return {
+            "data": {
+                "type": "campaign-values-report",
+                "attributes": {
+                    "statistics": [
+                        "click_rate",
+                        "click_to_open_rate",
+                        "clicks",
+                        "clicks_unique",
+                        "delivered",
+                        "delivery_rate",
+                        "open_rate",
+                        "opens",
+                        "opens_unique",
+                        "recipients",
+                        "unsubscribe_rate",
+                        "unsubscribe_uniques",
+                        "unsubscribes",
+                        ],
+                    "timeframe": {
+                        "key": "last_12_months"
+                    },
+                    "conversion_metric_id": "WcGvVS",
+                    "filter": f"equals(campaign_id,\"{context['campaign_id']}\")"
+                }
+            }
+        }
+
+    def backoff_wait_generator(self) -> t.Generator[float, None, None]:
+        def _backoff_from_headers(retriable_api_error):
+            response_headers = retriable_api_error.response.headers
+            return int(response_headers.get("Retry-After", 60))
+
+        return self.backoff_runtime(value=_backoff_from_headers)
+
+    def get_records(self, context: dict | None) -> t.Iterable[dict[str, t.Any]]:
+        # Only fetch records if the campaign has been sent
+        if context['campaign_status'] == 'Sent':
```

Comment:
I'll add a config to help limit the number of records to avoid reaching the daily limits which is currently 225 reports. https://developers.klaviyo.com/en/reference/query_campaign_values